lib/compat/wordpress-6.8/admin-bar.php

```diff
@@ -30,3 +30,18 @@ function gutenberg_wp_admin_bar_edit_site_menu( $wp_admin_bar ) {
 }
 
 add_action( 'admin_bar_menu', 'gutenberg_wp_admin_bar_edit_site_menu', 41 );
+
+/**
+ * Hide the admin bar in the classic theme site preview.
+ *
+ * @param bool $show_admin_bar Whether to show the admin bar.
+ * @return bool
+ */
+function gutenberg_hide_admin_bar_in_site_preview( $show_admin_bar ) {
+	if ( isset( $_GET['wp_site_preview'] ) && 1 === (int) $_GET['wp_site_preview'] ) {
+		return false;
+	}
+	return $show_admin_bar;
+}
+
+add_filter( 'show_admin_bar', 'gutenberg_hide_admin_bar_in_site_preview' );
```

Comment:
What's the reason for going in a slightly different route here instead of having the `wp_initialize_site_preview_hooks` method like core PR?